jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/FilesTaskV2.java

```diff
@@ -0,0 +1,193 @@
+      skaffoldFilesOutput.addInput(jibExtension.getExtraDirectory().getPath());
+    }
+
+    // Find project dependencies
+    Set<ProjectDependency> projectDependencies = findProjectDependencies(project);
+
+    Set<File> projectDependencyJars = new HashSet<>();
+    for (ProjectDependency projectDependency : projectDependencies) {
+      addProjectFiles(projectDependency.getDependencyProject());
+
+      // Keep track of project dependency jars for filtering out later
+      String configurationName = projectDependency.getTargetConfiguration();
+      if (configurationName == null) {
+        configurationName = "default";
+      }
+      Project dependencyProject = projectDependency.getDependencyProject();
+      for (Configuration targetConfiguration :
+          dependencyProject.getConfigurations().getByName(configurationName).getHierarchy()) {
+        for (PublishArtifact artifact : targetConfiguration.getArtifacts()) {
+          projectDependencyJars.add(artifact.getFile());
+        }
+      }
+    }
+
+    // Add SNAPSHOT, non-project dependency jars
+    for (File file : project.getConfigurations().getByName("runtime")) {
+      if (!projectDependencyJars.contains(file) && file.toString().contains("SNAPSHOT")) {
+        skaffoldFilesOutput.addInput(file.toPath());
+        projectDependencyJars.add(file); // Add to set to avoid printing the same files twice
+      }
+    }
+
+    // Print files
+    System.out.println(skaffoldFilesOutput.getJsonString());
+  }
+
+  /**
+   * Prints the locations of a project's build.gradle, settings.gradle, and gradle.properties.
+   *
+   * @param project the project
+   */
+  private void addGradleFiles(Project project) {
+    Path projectPath = project.getProjectDir().toPath();
+
+    // Print build.gradle
+    skaffoldFilesOutput.addBuild(project.getBuildFile().toPath());
+
+    // Print settings.gradle
+    if (project.getGradle().getStartParameter().getSettingsFile() != null) {
+      skaffoldFilesOutput.addBuild(
+          project.getGradle().getStartParameter().getSettingsFile().toPath());
+    } else if (Files.exists(projectPath.resolve(Settings.DEFAULT_SETTINGS_FILE))) {
+      skaffoldFilesOutput.addBuild(projectPath.resolve(Settings.DEFAULT_SETTINGS_FILE));
+    }
+
+    // Print gradle.properties
+    if (Files.exists(projectPath.resolve("gradle.properties"))) {
+      skaffoldFilesOutput.addBuild(projectPath.resolve("gradle.properties"));
+    }
+  }
+
+  /**
+   * Prints build files, sources, and resources associated with a project.
+   *
+   * @param project the project
+   */
+  private void addProjectFiles(Project project) {
+    JavaPluginConvention javaConvention =
+        project.getConvention().getPlugin(JavaPluginConvention.class);
+    SourceSet mainSourceSet =
+        javaConvention.getSourceSets().findByName(SourceSet.MAIN_SOURCE_SET_NAME);
+    if (mainSourceSet == null) {
+      return;
+    }
+
+    // Print build config, settings, etc.
+    addGradleFiles(project);
```

Comment:
I was a little confused by us calling `addGradleFiles` here and in the main task action (https://github.com/GoogleContainerTools/jib/pull/1492/files#diff-6459724c2453bb9a6d621ab67dd3c866R65). What do you think about making this method just deal with sources and resources and calling both `addGradlefile` and `addProjectSources` directly in the main taskaction method?

Reply:
I think the one in the main task action adds the build files for the root project in case you run the task on a sub-project, and the one in `addProjectFiles` adds the sub-project's build files.